client/apps/edit/components/content2/sections/header/controls/index.jsx

```diff
@@ -0,0 +1,73 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import ModalCover from './ModalCover.jsx'
+import LayoutControls from './LayoutControls.jsx'
+import VideoControls from './VideoControls.jsx'
+
+export default class HeaderControls extends Component {
+  static propTypes = {
+    article: PropTypes.object,
+    onChange: PropTypes.func.isRequired,
+    hero: PropTypes.object
+  }
+
+  state = {
+    isLayoutOpen: false,
+    isVideoOpen: false
+  }
+
+  toggleControls = (controlType) => {
+    const { isLayoutOpen, isVideoOpen } = this.state
+
+    switch (controlType) {
+      case 'layout': {
+        return this.setState({
+          isLayoutOpen: !isLayoutOpen,
+          isVideoOpen: false
+        })
+      }
+      case 'video': {
+        return this.setState({
+          isVideoOpen: !isVideoOpen,
+          isLayoutOpen: false
+        })
+      }
+      case 'close': {
+        return this.setState({
+          isVideoOpen: false,
+          isLayoutOpen: false
+        })
+      }
+    }
+  }
+
+  render () {
+    const { isLayoutOpen, isVideoOpen } = this.state
+    const { article, hero, onChange } = this.props
+    const isModalOpen = isLayoutOpen || isVideoOpen
+    const isBasicHero = hero.type === 'basic'
+
+    return (
+      <div className='edit-header__container'>
+        {isModalOpen &&
+          <ModalCover
+            onClick={() => this.toggleControls('close')}
+          />
+        }
+        {isBasicHero &&
+          <VideoControls
+            article={article}
+            isOpen={isVideoOpen}
+            onClick={() => this.toggleControls('video')}
+          />
+        }
+        <LayoutControls
+          hero={hero}
+          isOpen={isLayoutOpen}
+          onChange={onChange}
+          onClick={() => this.toggleControls('layout')}
+        />
+      </div>
+    )
```

Comment:
I started working the controls into the structure we discussed earlier today, but realized that it would probably be better to refactor out the controls into something more container-like so as to keep things decoupled from reaction / self-contained and a bit simpler. I wanted the <BasicHeader /> to just be a dumb rendering component for props and not have to worry about the outside world, children, etc. Would be open to refactoring further at a later point but given the short time frame this is good to go.

Reply:
This looks like an ok solution to me.